dear_petition/petition/api/viewsets.py

```diff
@@ -407,6 +427,26 @@ def assign_client_to_batch(self, request, pk):
         batch.save()
         batch.adjust_for_new_client_dob()
         return Response({"batch_id": batch.pk})
+    
+    @action(
+        detail=False,
+        methods=[
+            "post",
+        ],
+    )
+    def import_spreadsheet(self, request):
+        files = request.data.getlist("files")
```

Comment:
Consider try/except block to gracefully handle errors.  Also do you need to make the transaction atomic?

Reply:
I added an atomic block to the import_data method of the resource

Reply:
So does it handle it gracefully if user imports spreadsheet with unexpected format?  Or is try/except needed?